### PR TITLE
TPF: drop support

### PR DESCRIPTION
**Pull request by bagder**
There has been no TPF related changes done since September 2010 (commit
7e1a45e224e57) and since this is a platform that is relatively different
than many others (== needs attention), I draw the conclusion that this
build is broken since a long time.

### CODE REVIEW

**Comment by monnerat** [comment]
If you do that, this is also an open door for all the `CURL_DOES_CONVERSIONS` stuff removal, as it seems this is the only platform using it.
Just my 2 cents!

**Comment by bagder** [comment]
Oh wow, thanks for highlighting this fact. I think that's actually also another strong reason to suspect that the TPF build doesn't work anymore as the conversion stuff is fairly complicated and easy to mess up.

Also, a fairly big gain to be able to rip that out as this define complicates the logic in many places.

**Comment by monnerat** [comment]
Big slashing, isn't it ! :-)

**Comment by bagder** [comment]
Yes, that is a significant chunk. There might even be some more to fix after this when some flows have been simplified.

File: lib/setopt.c [hunk 1]
```diff
@@ -1644,22 +1644,9 @@ CURLcode Curl_vsetopt(struct Curl_easy *data, CURLoption option, va_list param)
...
-     * "Convert to network encoding" callback
-     */
-    data->set.convtonetwork = va_arg(param, curl_conv_callback);
-    break;
   case CURLOPT_CONV_FROM_UTF8_FUNCTION:
-    /*
-     * "Convert from UTF-8 encoding" callback
-     */
-    data->set.convfromutf8 = va_arg(param, curl_conv_callback);
+    /* removed */
```
Comment (jay) [inline lib/setopt.c:1649]: I think it would be better to remove them entirely (CURLE_UNKNOWN_OPTION), but if not you'll need /* FALLTHROUGH */

Comment (bagder) [reply lib/setopt.c:1649]: yeah, might just as well remove them completely

**Pull request closed without merging**